services/src/main/java/org/exoplatform/addons/gamification/storage/dao/GamificationHistoryDAO.java

```diff
@@ -408,13 +410,31 @@ public Long countAnnouncementsByChallenge(Long challengeId) {
     }
   }
 
-  public List<GamificationActionsHistory> findAllAnnouncementByChallenge(Long challengeId, int offset, int limit) {
-    TypedQuery<GamificationActionsHistory> query =
-                                                 getEntityManager().createNamedQuery("GamificationActionsHistory.findAllAnnouncementByChallenge",
-                                                                                     GamificationActionsHistory.class);
+  public List<GamificationActionsHistory> findAllAnnouncementByChallenge(Long challengeId,
+                                                                         int offset,
+                                                                         int limit,
+                                                                         RuleFilter ruleFilter) {
```

Comment:
no need to add the whole rule filter, just pass the period as a parameter.

Reply:
okay am on it